app/temporal/src/activities.ts

```diff
@@ -81,175 +88,150 @@ export function makeActivities(
+    }): Promise<void> => {
+      for (const patch of params.patches) {
+        const targetColumn = `data.${patch.column}`;
+        await database.mongoClient
+          .db(params.importerId)
+          .collection("data")
+          .updateOne(
+            { __sourceRowId: patch.rowId },
+            { $set: { [targetColumn]: patch.newValue } }
+          );
+      }
     },
     generateStats: async (params: {
-      bucket: string;
-      fileReference: string;
+      importerId: string;
       uniqueColumns: string[];
-      patches: DataSetPatch[];
     }): Promise<SourceFileStatsPerColumn> => {
       console.time("generate-stats");
-      const fileData = await fileStore.getFile(
-        params.bucket,
-        params.fileReference
-      );
-      const jsonData: DataSet = JSON.parse(fileData.toString());
-      const patchedData = applyPatches(jsonData, params.patches);
-      const stats = dataAnalyzer.getStats(patchedData, params.uniqueColumns);
+      const jsonData: DataSet = await database.mongoClient
+        .db(params.importerId)
+        .collection<DataSetRow>("data")
+        .find()
+        .toArray();
+      const stats = dataAnalyzer.getStats(jsonData, params.uniqueColumns);
       console.timeEnd("generate-stats");
       return stats;
     },
-    mergeChunks: async (params: {
-      bucket: string;
-      fileReferences: string[];
-      outputFileReference: string;
-    }) => {
-      let allJsonData: DataSet = [];
-      for (const fileReference of params.fileReferences) {
-        const fileData = await fileStore.getFile(params.bucket, fileReference);
-        allJsonData.push(...JSON.parse(fileData.toString()));
-      }
-      await fileStore.putFile(
-        params.bucket,
-        params.outputFileReference,
-        Buffer.from(JSON.stringify(allJsonData))
-      );
-    },
-    export: async (params: {
-      bucket: string;
-      fileReference: string;
-      patches: DataSetPatch[];
+    invokeCallback: async (params: {
+      importerId: string;
       callbackUrl: string;
-      exportFileReference: string;
-    }): Promise<string> => {
-      const fileData = await fileStore.getFile(
-        params.bucket,
-        params.fileReference
-      );
-      const allJsonData: DataSet = JSON.parse(fileData.toString());
-      const patchedData = applyPatches(allJsonData, params.patches ?? []);
-      await fileStore.putFile(
-        params.bucket,
-        params.exportFileReference,
-        Buffer.from(JSON.stringify(patchedData))
-      );
+    }): Promise<void> => {
       const host = process.env.API_URL ?? "http://localhost:3000";
-      const downloadUrl = `${host}/api/download/${params.bucket}`;
+      const downloadUrl = `${host}/api/download/${params.importerId}`;
       fetch(params.callbackUrl, {
         method: "POST",
         body: downloadUrl,
       });
-      return params.exportFileReference;
+    },
+    getDataCount: async (params: { importerId: string }): Promise<number> => {
```

Comment:
Warum einzeln und nicht in der Stats activity?

Reply:
good point - done